### DISCUSSION TITLE
How to provide my own API key for the Marginalia search engine?

### DISCUSSION THREAD

**Discussion by justinclift** [searxng/searxng#2412] [Q&A] [answered]
At present, the Marginalia search engine is using their demo API key ('public'), which isn't really useful.

I'd like it to use my own API key (i.e. one issued to me by Marginalia), but I'm not sure how/where to put it.

Any ideas? :smile:

**Comment by allendema** [comment]
You would have to change the "public" part for marginalia in [settings.yml with your key](https://github.com/searxng/searxng/blob/d867bf17e6d2f9a7c83c9a1ffafda5184a24c0e3/searx/settings.yml#L1482).

See more:
https://www.marginalia.nu/marginalia-search/api/

**Comment by dalf** [comment]
Reading https://news.ycombinator.com/item?id=35874640 it would be nice to:
* either disable the engine by default and provide a way to specify the key.
* or at least write a warning in settings.yml and how to provide the API key.

**Reply by return42** [reply, marked as answer]
@dalf Thank you very much for the note. I have removed the engine from the configuration for now, see https://github.com/searxng/searxng/pull/2489
